components/ConnectWalletButton.tsx

```diff
@@ -38,14 +38,13 @@ const ConnectWalletButton = (props) => {
     providerUrl,
   ])
 
-  const [useDevnet, setUseDevnet] = useState(false)
-  const router = useRouter()
+  const [useDevnet, setUseDevnet] = useLocalStorageState('false')
   const handleToggleDevnet = () => {
     setUseDevnet(!useDevnet)
     if (useDevnet) {
-      router.push(`${window.location.pathname}`)
+      window.location.href = `${window.location.pathname}`
```

Comment:
When selected from this toggle button, the cluster should always be present in the URL to assert priority over possible existing local storage preference.

Reply:
I'm not sure I understand what you mean. The part of the codebase you're pointing to is when there isn't a cluster specified (ie mainnet). The part that's confusing me is what you mean by priority. Are you saying it should be more explicit? For example:

```
window.location.href = `${window.location.pathname}?cluster=mainnet`
```